CoreImportExportPlugins/src/au/gov/asd/tac/constellation/plugins/importexport/delimited/parser/PCAPImportFileParser.java

```diff
@@ -0,0 +1,173 @@
+                    String protocol = "";
+                    String ipVersion = "";
+                    Buffer buffer = packet.getPayload();
+                    String length = Integer.toString(buffer.capacity());
+                         
+                    try {               
+                        
+                        if (packet.hasProtocol(Protocol.TCP)) {
+                            protocol = "TCP";
+                            TCPPacket tcpPacket = (TCPPacket) packet.getPacket(Protocol.TCP);
+                            sourcePort = Integer.toString(tcpPacket.getSourcePort());
+                            destinationPort = Integer.toString(tcpPacket.getDestinationPort());
+                            ipPacket = (IPPacket) tcpPacket.getParentPacket();
+                        } else if (packet.hasProtocol(Protocol.UDP)) {
+                            protocol = "UDP";
+                            UDPPacket udpPacket = (UDPPacket) packet.getPacket(Protocol.UDP);
+                            sourcePort = Integer.toString(udpPacket.getSourcePort());
+                            destinationPort = Integer.toString(udpPacket.getDestinationPort());
+                            ipPacket = (IPPacket) udpPacket.getParentPacket();
+                        }
+                        
+                        if (ipPacket != null) {
+                            ipVersion = "IPv" + Integer.toString(ipPacket.getVersion());
+                            sourceIP = ipPacket.getSourceIP();
+                            sourceType = ipVersion + " Address";
+                            destinationIP = ipPacket.getDestinationIP();
+                            destinationType = ipVersion + " Address";
+                        }
+                        
+                        LOGGER.log(Level.INFO, "{0}, {1}, {2}, {3}, {4}, {5}, {6}, {7}, {8}, {9}, {10}", new Object[]{frame, formatter.format(timestamp), sourceIP, sourcePort, sourceType, destinationIP, destinationPort, destinationType, protocol, length });
+                        final String[] row = {frame, formatter.format(timestamp), sourceIP, sourcePort, sourceType, destinationIP, destinationPort, destinationType, protocol, length};
+                        allResults.add(row);
+                    } catch (AssertionError e) {
+                        LOGGER.log(Level.INFO, "ASSERTION THROWN 1 {0}", e.toString());
+                    }
+                    catch (Exception e) {
+                        LOGGER.log(Level.INFO, "EXCEPTION THROWN 1 {0}", e.toString());
+                    }
+                    
+                    frameCounter ++;
+                    return (frameCounter <= limit || limit == 0);
+                }
+            });
+            
+            return allResults;
+        } catch (AssertionError e) {
+                        LOGGER.log(Level.INFO, "ASSERTION THROWN 2 {0}", e.toString());
+          return allResults;
+        } catch (Exception ex) {
```

Comment:
final Exception (also, can this be more specific?)

Reply:
done